.github/workflows/solutions.yml

```diff
@@ -0,0 +1,23 @@
+name: Solutions
+
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+
+jobs:
+  test:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions/setup-node@v3
+        with:
+          node-version: "16"
+          cache: "npm"
+      - run: npm ci
+      - name: Run all solutions tests
+        run: |
+          chmod +x ./scripts/check-solutions.sh
```

Comment:
Oh, and you can run this locally and check in the file with permissions changed. People will want it to be `+x` executable locally so they can run it.

Reply:
would it make any difference if cloned ? 🤔

Reply:
Oh It looks like it would 🤯 , I ran it and it showed as a file change

Reply:
Didn't know that was possible 🤯

Reply:
Git should respect the permissions the file is checked in with. When I run it locally:

```plaintext
~/repos/learning-typescript-projects$ chmod +x ./scripts/check-solutions.sh 
~/repos/learning-typescript-projects$ git add -A
~/repos/learning-typescript-projects$ git diff HEAD
diff --git a/scripts/check-solutions.sh b/scripts/check-solutions.sh
old mode 100644
new mode 100755
~/repos/learning-typescript-projects$ 
```